core/app/models/spree/product.rb

```diff
@@ -105,7 +105,7 @@ class Product < Spree::Base
       validates :price, if: proc { Spree::Config[:require_master_price] }
     end
 
-    validates :slug, length: { minimum: 3 }, allow_blank: true, uniqueness: true
+    validates :slug, allow_blank: true, uniqueness: true
```

Comment:
[14] pry(main)> p.slug = ''
=> ""
[15] pry(main)> p.valid?
  Spree::Preference Load (0.4ms)  SELECT  "spree_preferences".* FROM "spree_preferences" WHERE "spree_preferences"."key" = $1 LIMIT $2  [["key", "spree/app_configuration/require_master_price"], ["LIMIT", 1]]
=> false
[16] pry(main)> p.slug = nil
=> nil
[17] pry(main)> p.valid?
  Spree::Product Exists (0.6ms)  SELECT  1 AS one FROM "spree_products" INNER JOIN "friendly_id_slugs" ON "friendly_id_slugs"."sluggable_id" = "spree_products"."id" AND "friendly_id_slugs"."deleted_at" IS NULL AND "friendly_id_slugs"."sluggable_type" = $1 WHERE ("spree_products"."id" != 2) AND "friendly_id_slugs"."deleted_at" IS NULL AND (sluggable_id <> 2) AND "spree_products"."slug" = 'asd-product' LIMIT $2  [["sluggable_type", "Spree::Product"], ["LIMIT", 1]]
  Spree::Product Exists (0.5ms)  SELECT  1 AS one FROM "spree_products" INNER JOIN "friendly_id_slugs" ON "friendly_id_slugs"."sluggable_id" = "spree_products"."id" AND "friendly_id_slugs"."deleted_at" IS NULL AND "friendly_id_slugs"."sluggable_type" = $1 WHERE ("spree_products"."id" != 2) AND "friendly_id_slugs"."deleted_at" IS NULL AND (sluggable_id <> 2) AND ("friendly_id_slugs"."sluggable_type" = 'Spree::Product' AND "friendly_id_slugs"."slug" = 'asd-product') LIMIT $2  [["sluggable_type", "Spree::Product"], ["LIMIT", 1]]
  Spree::Product Exists (0.4ms)  SELECT  1 AS one FROM "spree_products" INNER JOIN "friendly_id_slugs" ON "friendly_id_slugs"."sluggable_id" = "spree_products"."id" AND "friendly_id_slugs"."deleted_at" IS NULL AND "friendly_id_slugs"."sluggable_type" = $1 WHERE ("spree_products"."id" != 2) AND "friendly_id_slugs"."deleted_at" IS NULL AND (sluggable_id <> 2) AND "spree_products"."slug" = 'asd-product' LIMIT $2  [["sluggable_type", "Spree::Product"], ["LIMIT", 1]]
  Spree::Product Exists (0.5ms)  SELECT  1 AS one FROM "spree_products" INNER JOIN "friendly_id_slugs" ON "friendly_id_slugs"."sluggable_id" = "spree_products"."id" AND "friendly_id_slugs"."deleted_at" IS NULL AND "friendly_id_slugs"."sluggable_type" = $1 WHERE ("spree_products"."id" != 2) AND "friendly_id_slugs"."deleted_at" IS NULL AND (sluggable_id <> 2) AND ("friendly_id_slugs"."sluggable_type" = 'Spree::Product' AND "friendly_id_slugs"."slug" = 'asd-product') LIMIT $2  [["sluggable_type", "Spree::Product"], ["LIMIT", 1]]
  Spree::Preference Load (0.3ms)  SELECT  "spree_preferences".* FROM "spree_preferences" WHERE "spree_preferences"."key" = $1 LIMIT $2  [["key", "spree/app_configuration/require_master_price"], ["LIMIT", 1]]
  Spree::Product Exists (0.3ms)  SELECT  1 AS one FROM "spree_products" WHERE "spree_products"."slug" = $1 AND "spree_products"."deleted_at" IS NULL AND ("spree_products"."id" != $2) LIMIT $3  [["slug", "asd-product-c7a2f169-89c1-48a2-a378-2de2cce099af"], ["id", 2], ["LIMIT", 1]]
=> true
[18] pry(main)> p.slug
=> "asd-product-c7a2f169-89c1-48a2-a378-2de2cce099af"